src/robothub_oak/live_view.py

```diff
@@ -0,0 +1,290 @@
+    return color_stream_handle
+
+
+def _publish_data(stream_handle: robothub_core.StreamHandle,
+                  h264_frame,
+                  rectangles: List[BoundingBox],
+                  rectangle_labels: List[str],
+                  texts: List[VisText],
+                  lines: List[VisLine],
+                  frame_width: int,
+                  frame_height: int):
+    timestamp = int(time.perf_counter_ns() / 1e6)
+    metadata = {
+        "platform": "robothub",
+        "frame_shape": [frame_height, frame_width],
+        "config": {
+            "output": {
+                "img_scale": 1.0,
+                "show_fps": False,
+                "clickable": True
+            },
+            "detection": {
+                "thickness": 1,
+                "fill_transparency": 0.05,
+                "box_roundness": 0,
+                "color": [0, 255, 0],
+                "bbox_style": 0,
+                "line_width": 0.5,
+                "line_height": 0.5,
+                "hide_label": False,
+                "label_position": 0,
+                "label_padding": 10
+            },
+            'text': {
+                'font_color': [255, 255, 0],
+                'font_transparency': 0.5,
+                'font_scale': 1.0,
+                'font_thickness': 2,
+                'bg_transparency': 0.5,
+                'bg_color': [0, 0, 0]
+            }
+        },
+        "objects": [
+            {
+                "type": "detections",
+                "detections": []
+            }
+        ]
+    }
+
+    # Bounding boxes
+    for roi, label in zip(rectangles, rectangle_labels):
+        xmin, ymin, xmax, ymax = roi
+        metadata['objects'][0]['detections'].append(
+            {'bbox': [xmin, ymin, xmax, ymax], 'label': label, 'color': [0, 255, 255]}
+        )
+
+    # Texts
+    for text in texts:
+        metadata["objects"].append(text.prepare().serialize())
+
+    # Lines
+    for line in lines:
+        metadata["objects"].append(line.prepare().serialize())
+
+    # Publish
+    stream_handle.publish_video_data(bytes(h264_frame), timestamp, metadata)
+
+
+class LiveView:
+    def __init__(self,
+                 title: str,
+                 unique_key: str,
+                 device_mxid: str,
+                 frame_width: int,
+                 frame_height: int):
+        """
+        :param title: Name of the Live View.
+        :param unique_key: Live View identifier.
+        :param device_mxid: MXID of the device that is streaming the Live View.
+        :param frame_width: Frame width.
+        :param frame_height: Frame height.
+        """
+        self.frame_width = frame_width
+        self.frame_height = frame_height
+        self.device_mxid = device_mxid
+        self.unique_key = unique_key
+        self.title = title
+
+        self.stream_handle = _create_stream_handle(camera_serial=device_mxid, unique_key=unique_key, title=title)
```

Comment:
I know we need unique key here, i dont know why, but until we fix this, i say we forget about the unique_key in this abstraction. Lets give the user one clear way how to create the live view